plugin/pkg/scheduler/algorithm/scheduler_interface.go

```diff
@@ -49,8 +49,9 @@ type ScheduleAlgorithm interface {
 	Schedule(*v1.Pod, NodeLister) (selectedMachine string, err error)
 	// Preempt receives scheduling errors for a pod and tries to create room for
 	// the pod by preempting lower priority pods if possible.
-	// It returns the node where preemption happened, a list of preempted pods, and error if any.
-	Preempt(*v1.Pod, NodeLister, error) (selectedNode *v1.Node, preemptedPods []*v1.Pod, err error)
+	// It returns the node where preemption happened, a list of preempted pods, a
+	// list of pods whose nominated node name should be removed, and error if any.
```

Comment:
Add to the comment (wherever else makes more sense) why you remove nominated node name from the lower-priority pods.

Reply:
I added a comment to Preempt() in generic_scheduler.go where it collects and returns pods for removal of their nomination.